### PR TITLE
Feature/star aln recipe split

### PR DESCRIPTION
**Pull request by jemten**
### This PR fixes:
- Align all sample fastq files in one STAR command if the sequence type
  is the same
- STAR adds read group on the fly
- Introduces optical duplicate distance for picardtools markduplicates.
  Default 2500

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: lib/MIP/Set/Analysis.pm [hunk 1]
```diff
@@ -475,4 +477,73 @@ sub set_ase_chain_recipes {
...
+            strict_type => 1,
+        },
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    use MIP::Recipes::Analysis::Star_aln qw{ analysis_star_aln analysis_star_aln_mixed };
+    use MIP::Sample_info qw{ get_sequence_run_type };
+
+    ## Get sequence run modes
```
Comment (henrikstranneheim) [inline lib/MIP/Set/Analysis.pm]:
```suggestion
    ## Get sequence run types
```

File: t/analysis_star_aln_mixed.t [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+);
+@{ $parameter{cache}{order_recipes_ref} } = ($recipe_name);
+$parameter{$recipe_name}{outfile_suffix} = q{.bam};
+
+my %sample_info = (
+    sample => {
+        $sample_id => {
+            file => {
+                ADM1059A1_161011_TestFilev2_GAGATTCC_lane1 => {
```
Comment (henrikstranneheim) [inline t/analysis_star_aln_mixed.t]: Maybe you should add a single-end fastq as the recipe is intended to run mixed cases

File: lib/MIP/Recipes/Analysis/Star_aln.pm [hunk 3]
```diff
@@ -141,12 +142,383 @@ sub analysis_star_aln {
...
+    ## Perform per single-end or read pair
+  INFILE_PREFIX:
+    while ( my ( $infile_index, $infile_prefix ) =
+        each @{ $infile_lane_prefix_href->{$sample_id} } )
+    {
+
+        # Collect paired-end or single-end sequence run mode
+        my $sequence_run_mode =
+          $sample_info_href->{sample}{$sample_id}{file}{$infile_prefix}
+          {sequence_run_type};
```
[the remark this answers is not in the document]
Comment (jemten) [reply lib/MIP/Recipes/Analysis/Star_aln.pm]: Ah! I forgot about that sub

File: lib/MIP/Recipes/Analysis/Star_aln.pm [hunk 4]
```diff
@@ -141,12 +142,383 @@ sub analysis_star_aln {
...
+                infile_prefix    => $infile_prefix,
+                platform         => $active_parameter_href->{platform},
+                sample_id        => $sample_id,
+                sample_info_href => $sample_info_href,
+            }
+        );
+
+        ## Construct read group line;
+        my @rg_elements = map { uc $_ . $COLON . $rg{$_} } qw{ id lb pl pu sm };
+        push @read_groups, join $SPACE, @rg_elements;
```
[the remark this answers is not in the document]
Comment (jemten) [reply lib/MIP/Recipes/Analysis/Star_aln.pm]: Will look into it :)

File: t/analysis_star_aln_mixed.t (hunk 2 above)
Comment (jemten) [reply t/analysis_star_aln_mixed.t]: Good catch

**Comment by jemten** [comment]
I added the read group header sub to the BWA recipe as well. Not as smooth as in star_aln perhaps but take a look and let me know what you think

File: lib/MIP/Sample_info.pm [hunk 5]
```diff
@@ -204,6 +205,77 @@ sub get_read_group {
     return %rg;
 }
 
+sub get_rg_header_line {
+
+## Function : Builds hash with read group headers
```
Comment (henrikstranneheim) [inline lib/MIP/Sample_info.pm]:
```suggestion
## Function : Builds line using read group headers
```

File: lib/MIP/Sample_info.pm [hunk 6]
```diff
@@ -204,6 +205,77 @@ sub get_read_group {
...
+## Returns  : %read_group
```
Comment (henrikstranneheim) [inline lib/MIP/Sample_info.pm]:
```suggestion
## Returns  : $rg_header_line
```

File: lib/MIP/Sample_info.pm [hunk 7]
```diff
@@ -204,6 +205,77 @@ sub get_read_group {
...
+## Arguments: $infile_prefix    => Name of Fastq file minus read direction information
```
Comment (henrikstranneheim) [inline lib/MIP/Sample_info.pm]:
```suggestion
## Arguments: $infile_prefix    => Name of fastq file minus read direction information
```

**Pull request merged**